libr/include/r_types_overflow.h

```diff
@@ -0,0 +1,53 @@
+#ifndef R_TYPES_OVERFLOW_H
+#define R_TYPES_OVERFLOW_H
+
+// TODO: Use CLANG/GCC builtins if available: __builtin_mul_overflow
+
+// ADD
+// if ((x > 0) && (a > INT_MAX - x)) /* `a + x` would overflow */;
+// if ((x < 0) && (a < INT_MIN - x)) /* `a + x` would underflow */;
```

Comment:
it seems like you are not considering the `INT_MIN` part in all the signed overflow check.
For example:
```C
#define SSZT_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > SSIZE_MAX - (x))) || (((x) < 0) && (a) < - (x)))
```
this should probably be,
```C
#define SSZT_ADD_OVFCHK(a,x) ((((x) > 0) && ((a) > SSIZE_MAX - (x))) || (((x) < 0) && (a) < SSIZE_MIN - (x)))
```
This is missed in many other checks and I think it's the reason for https://github.com/radareorg/radare2/pull/17216/files#diff-34737e40da2df87d09d8a4ffbbaf4602R24

Reply:
there's no SSIZE_MIN, but i can invent it

Reply:
if we don't have SSIZE_MIN this kind of check is not going to work :(

Reply:
done